packages/gatsby/cache-dir/loader.js

```diff
@@ -146,13 +151,25 @@ export class BaseLoader {
     // }
     this.pageDb = new Map()
     this.inFlightDb = new Map()
+
+    this.staticQueryDb = new Map()
+
+    this.inFlightPromisesByUrlDb = new Map()
+
     this.pageDataDb = new Map()
     this.prefetchTriggered = new Set()
     this.prefetchCompleted = new Set()
     this.loadComponent = loadComponent
     setMatchPaths(matchPaths)
   }
 
+  memoizedGet(url) {
+    if (!this.inFlightPromisesByUrlDb.has(url)) {
+      this.inFlightPromisesByUrlDb.set(url, doFetch(url, `GET`))
```

Comment:
Need to clean inFlight promise after it's done

```suggestion
      const fetchPromise = doFetch(url, `GET`)
      this.inFlightPromisesByUrlDb.set(url, fetchPromise)
      fetchPromise.then(() => {
        this.inFlightPromisesByUrlDb.delete(url)
      })
```

Reply:
Yup, done